### PR TITLE
SNOW-176219 augmenting external logger to be used for every log line

### PR DESCRIPTION
**Pull request by sfc-gh-smirzaei**
Currently the ISFLogger (the external logger) is only used in ODBC and for put/get methods and not for all log lines. With this change I augmented that so logLineVA method of the external logger is called for every log line.
See how it's used in https://github.com/snowflakedb/snowflake/pull/27804

### CODE REVIEW

**Review by sfc-gh-sjensen — COMMENTED** [review]
LGTM. But someone from client team should sign off. 
Is there a way to add tests?

**Comment by sfc-gh-smirzaei** [comment]
@sfc-gh-vreddy @sfc-gh-kwagner can you please review this?

File: lib/logger.c [hunk 1]
```diff
@@ -97,6 +98,21 @@ void log_log(int level, const char *file, int line, const char *ns,
...
     va_start(args, fmt);
+
+    if (getExternalLogger()){
+      // va_list can only be consumed once. Make a copy so we can use it again
+      va_list copy;
+      va_copy(copy, args);
+
+      // Add the line (since logLineVA doesn't take line)
+      char linedFmt[strlen(fmt) + 10 /*max num of digits for line*/ + 3];
+      sprintf(linedFmt, "%d: %s", line, fmt);
```
[the remark this answers is not in the document]
Comment (sfc-gh-kwagner) [reply lib/logger.c:109]: Not really. Maybe make the 10 a named constant so we have fewer magic numbers

[the remark this answers is not in the document]
Comment (sfc-gh-smirzaei) [reply lib/logger.c:111]: Done

**Review by sfc-gh-kwagner — APPROVED** [review]
Looks good to me. Why all the file permission changes? Were those included on purpose?

**Comment by sfc-gh-smirzaei** [comment]
> Looks good to me. Why all the file permission changes? Were those included on purpose?

No, I didn't touch those and they are identical between the two branch so I don't know they are marked as changed!

**Pull request merged**